float32 flag
float32 x_err
float32 y_err
float32 leftORright